project-saarah/scenes/add-product-item/AddProductItemModels.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  AddProductItemModels.swift
+//  project-saarah
+//
+//  Created by Guilherme Colombini on 17/10/19.
+//  Copyright (c) 2019 Bruno Rocha. All rights reserved.
+//
+
+import Foundation
+
+enum AddProductItem {
+	struct AddItemForm {
+		let quantity: Double
+		let price: Double
+		let expirationDate: String
+	}
+	// MARK: Use cases
+	enum AddItem {
+		struct Request {
+			let addItemForm: AddItemForm
+		}
+		struct Response {
+			let apiMessage: String
+		}
+		struct ViewModel {
+			struct AddItemViewModel {
+				let success: Bool
```

Comment:
In this case 'success' should be an property of your Response, not of your view model.

Reply:
I use this property to dismiss or not a view controller, since this property is read only in the view controller makes sense stay in the view model.